classes/travel/TravelManager.php

```diff
@@ -287,6 +291,12 @@ public function fetchNearbyPlayers(): array {
                     $can_attack = true;
                 }
             }
+            // additionally, if player is jonin and target is chuunin
+            $is_protected = false;
+            if (User::isProtectedByAlly($this->system, user_id: $user['user_id']) && $this->user->rank_num == 4) {
```

Comment:
This is going to trigger <number of nearby players> database queries on every scout API request and we do those for every player viewing the travel page every 400ms, so this could become, worst case if you have 30 players online who all can see each other viewing scout page - 2,250 extra queries per second hitting the database.

I think we need to make this a batch query